integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/ContainerSuite.java

```diff
@@ -124,12 +131,46 @@ public HiveContainer getHiveContainer() {
   // Let containers assign addresses in a fixed subnet to avoid `mac-docker-connector` needing to
   // refresh the configuration
   private static Network createDockerNetwork() {
+    DockerClient dockerClient = DockerClientFactory.instance().client();
+
+    // Remove the `gravitino-ci-network` if it exists
+    boolean networkExists =
+        dockerClient.listNetworksCmd().withNameFilter(NETWORK_NAME).exec().stream()
+            .anyMatch(network -> network.getName().equals(NETWORK_NAME));
+    if (networkExists) {
+      RemoveNetworkCmd removeNetworkCmd = dockerClient.removeNetworkCmd(NETWORK_NAME);
+      removeNetworkCmd.exec();
+    }
+
+    // Check if the subnet of the network conflicts with `gravitino-ci-network`
+    List<com.github.dockerjava.api.model.Network> networks = dockerClient.listNetworksCmd().exec();
+
+    for (com.github.dockerjava.api.model.Network network : networks) {
+      List<Config> ipamConfigs = network.getIpam().getConfig();
+      for (Config ipamConfig : ipamConfigs) {
+        try {
+          if (ipRangesOverlap(ipamConfig.getSubnet(), CONTAINER_NETWORK_SUBNET)) {
+            throw new RuntimeException(
```

Comment:
I'm wondering if we can restart the docker containers when there are conflicts in the port range to avoid failure directly.

Reply:
The reason for the Docker Network Pool overlap is because the IP range requested by Gravition is already occupied by other services in Docker, so restarting Docker does not solve the problem.
So I added IP range conflict detection and alerts in this PR.